File: Array&String/CPP/split_a_string_in_balanced_strings.cpp
#include <string>
using namespace std;

class Solution {
public:
    int balancedStringSplit(string s) {
        int l = 0, r = 0, res = 0;
        for (char w : s) {
            if (w == 'L') l++;
            else r++;

            if (l == r) {
                res++;
                l = 0;
                r = 0;
            }
        }
        return res;
    }
};

File: Array&String/split_a_string_in_balanced_strings.py
class Solution:
    def balancedStringSplit(self, s: str) -> int:
        ans, l, r = 0, 0, 0
        for c in s:
            if c == 'L': l += 1
            else: r += 1

            if l == r:
                ans += 1
                l, r = 0, 0
        return ans